src/PrestaShopBundle/Translation/DTO/DomainTranslation.php

```diff
@@ -0,0 +1,238 @@
+        if (!array_key_exists($messageTranslation->getKey(), $this->messageTranslations)) {
+            // The missing translations are placed on top
+            if (!$messageTranslation->isTranslated()) {
+                $this->messageTranslations = [$messageTranslation->getKey() => $messageTranslation] + $this->messageTranslations;
+            } else {
+                $this->messageTranslations[$messageTranslation->getKey()] = $messageTranslation;
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return MessageTranslation[]
+     */
+    public function getMessageTranslations(): array
+    {
+        return $this->messageTranslations;
+    }
+
+    public function getTranslationsCount(): int
+    {
+        return count($this->messageTranslations);
+    }
+
+    public function getMissingTranslationsCount(): int
+    {
+        $missingTranslations = 0;
+        foreach ($this->messageTranslations as $messagesTranslation) {
+            if (!$messagesTranslation->isTranslated()) {
+                ++$missingTranslations;
+            }
+        }
+
+        return $missingTranslations;
+    }
+
+    /**
+     * Builds the domain metadata tree.
+     *
+     * Returns a structure like this:
+     *
+     * ```
+     * [
+     *     '__metadata' => [
+     *         'count' => 11,
+     *         'missing_translations' => 5
+     *     ],
+     *     'Admin' => [
+     *         '__metadata' => [
+     *             'count' => 11,
+     *             'missing_translations' => 5
+     *         ],
+     *         'Foo' => [
+     *             '__metadata' => [
+     *                 'count' => 4,
+     *                 'missing_translations' => 3
+     *             ],
+     *             'Bar' => [
+     *                 '__metadata' => [
+     *                     'count' => 2,
+     *                     'missing_translations' => 1
+     *                 ],
+     *             ],
+     *             'Baz' => [
+     *                 '__metadata' => [
+     *                     'count' => 2,
+     *                     'missing_translations' => 2
+     *                 ],
+     *             ],
+     *         ],
+     *         'Plop' => [
+     *             '__metadata' => [
+     *                 'count' => 7,
+     *                 'missing_translations' => 2
+     *             ],
+     *             'Foo' => [
+     *                 '__metadata' => [
+     *                     'count' => 2,
+     *                     'missing_translations' => 0
+     *                 ],
+     *             ],
+     *             'Bar' => [
+     *                 '__metadata' => [
+     *                     'count' => 3,
+     *                     'missing_translations' => 1
+     *                 ],
+     *             ],
+     *         ],
+     *     ],
+     * ];
+     * ```
+     *
+     * @param array $tree
+     *
+     * @return array
+     */
+    public function getTree(array &$tree): array
```

Comment:
```suggestion
    public function getTree(array &$tree): array
```
A getter that returns a tree, given a tree 🤔  ..? I'm confused 😅

Reply:
Same :sweat_smile:

Reply:
It's because we build the Tree by merging DomainTranslations trees 
If the domain is a subDomain, it must be inserted inside the tree of its parent

Maybe I can rename the method ...